src/main/resources/react/src/redux/Utils/toggleDisableButtonAuth.js

```diff
@@ -0,0 +1,9 @@
+export const toggleButtonLog = ( state ) => {
+	const { logEmail, logPassword } = state;
+	return (logEmail.value !== '' && logEmail.error === '' && logPassword.value !== '' && logPassword.error === '');
+};
+
+export const toggleButtonReg = ( state ) => {
+	const { regEmail, regNick, regPassword, regConfirm, regCheckbox } = state;
+	return (regEmail.value !== '' && regEmail.error === '' && regNick.value !== '' && regNick.error === '' && regPassword.value !== '' && regPassword.error === '' && regConfirm.value !== '' && regConfirm.error === '' && regCheckbox);
```

Comment:
Why don't you create a function that would get parameter and check it's **.value** and **.error** and return result of both those checks?
The code would be much shorter.

Reply:
thank you